File: StoreAPI/API/Data/Models/Product.cs
using System;
using System.Collections.Generic;

namespace API.Data.Models;

public partial class Product
{
    public int Id { get; set; }

    public string Name { get; set; } = null!;

    public string Description { get; set; } = null!;

    public decimal Price { get; set; }

    public int CategoryId { get; set; }

    public int ProductStateId { get; set; }

    public int Stock { get; set; }

    public virtual Category Category { get; set; } = null!;

    public virtual ICollection<OrderProduct> OrderProduct { get; set; } = new List<OrderProduct>();

    public virtual ProductState ProductState { get; set; } = null!;
}

File: StoreAdmUI/src/app/models/product.ts
import { Category } from "./category";
import { ProductState } from "./productState";

export interface Product {
    id: number;
    name: string;
    description: string;
    stock: number;
    price: number;
    categoryId: number;
    productStateId: number;
    cateogry: Category;
    productState: ProductState;
}